src/ex2/PokemonClient.js

```diff
@@ -0,0 +1,66 @@
+export default class PokemonClient {
+  constructor() {
+    this.API_BASE = "https://pokeapi.co/api/v2";
+  }
+
+  async fetchAllPokemons() {
+    try {
+      const pokemons = await fetch(
+        `${this.API_BASE}/pokemon?limit=100000&offset=0`
+      );
+      const pokemonsJson = await pokemons.json();
+
+      return pokemonsJson.results;
+    } catch {
+      (err) => console.log(err);
+    }
+  }
+
+  async fetchPokemonNameById(id) {
+    try {
+      const pokemonsList = await this.fetchAllPokemons();
```

Comment:
no need to fetch all pokemons here!
`https://pokeapi.co/api/v2/pokemon/1` will fetch pokemon with ID 1,
same for ` https://pokeapi.co/api/v2/pokemon/pikachu` to fetch pokemon by ID

Reply:
Hi, don't know if you saw my comment there but I wrote that I found out that fetching all Pokemons and then query them by ID is a lot faster then trying to fetch each of them by id. Do you still prefer to change it?